### ISSUE TITLE
ACL `whoami` command

### ISSUE THREAD

**Issue by Indigenuity** [hashicorp/consul#6471]
#### Feature Description

The first step when troubleshooting issues with credentials or permissions is to confirm what credentials/identity you're actually using.  Systems have commands like `whoami` or `aws sts get-caller-identity` in order to find this information.  I can imagine a similar command for ACLs in consul to reveal the AccessorID of whatever token is in use by agents or services or whatever:
```
$ consul acl whoami default
AccessorID:       5fcbd554-8f09-62d7-233b-217221dd3db3
Description:      My Agent Token

$ consul acl whoami replication
AccessorID:       5fcbd554-totally-62d7-different-217221dd3db3
Description:       My Replication Token

$ consul acl whoami service counting
AccessorID:       5fcbd554-totally-62d7-different-217221dd3db3
Description:      My Service Token
```

Ideally these `AccessorID`s would also be logged whenever a service or agent starts using a token.  You can't really audit a system that doesn't track how its entities authenticate.

Of course, I'm going off the assumption that `AccessorID` isn't a secret.  Correct me if I'm wrong there.

#### Use Case(s)

Imagine yourself troubleshooting a problem where a service suddenly doesn't have the right permissions after a machine restart.  Is it a token problem?  Is it an intentions problem? You can vaguely remember someone creating a new service token last week with a new policy, but that person is now OOO.   The same service is running correctly on another box, but there's no token in the service definition file.  So you don't even know what token is in use! Aside from the troubleshooting ahead of you, you can imagine your boss asking about auditing because you can't even find in the logs what service is authenticating using what token!

Basically, the troubleshooting use-cases mostly beg the question:

> What credentials are in use?

**Comment by s-christoff** [comment]
Hey @Indigenuity - we currently have something similar to this which you can read about [here](https://www.consul.io/docs/commands/acl/token/read.html#self). I hope this resolves your issue, if not go ahead and comment back and I can reopen this.

**Comment by Indigenuity** [comment]
That definitely helps when debugging tokens used by the agent itself!  The `-self` flag basically functions as `whoami` for the agent.  I don't see how to use that `read` command to figure out what token is in use by a service though.  Perhaps I'm missing something in that doc though?
